File: StackOverflow/stack_overflow.c
#include<stdio.h>

int fonction(int n) {
	int f=0;
	
	if(n<=0) {
		f=0;
	}
	if(n>=1) {
		f=1+(fonction(n-1));
	}
	return f;
}

int main() {
	printf("f(1000)=%d",fonction(1000));
	printf("f(1000000)=%d",fonction(1000000));
}

File: StackOverflow/stackoverflow.java
public class stackoverflow {

	public static int fonction(int n){
		int f=0;

		if(n<=0){
			f=0;
		}
		if(n>=1){
			f=1+(fonction(n-1));
		}
		return f;
	}

	public static void main (String args[]){
		fonction(1000);
		fonction(1000000);

	}


}
